tests/mpls/test_mpls.py

```diff
@@ -0,0 +1,594 @@
+ 
+    # get the list of TOR/SPINE ports
+    for dut_port, neigh in mg_facts['minigraph_neighbors'].items():
+        if not port_up(duthost, dut_port): 
+            continue
+        port_id = mg_facts['minigraph_port_indices'][dut_port]
+        if 'T0' in neigh['name']:
+            tor_ports[port_id]=dut_port
+            tor_ports_ids.append(port_id)
+        elif 'T2' in neigh['name']:
+            spine_ports[port_id]=dut_port
+            spine_ports_ids.append(port_id)
+    
+    logger.info('tor_ports: {}'.format(tor_ports))
+    logger.info('spine_ports: {}'.format(spine_ports))    
+        
+    for blk in mg_facts['minigraph_interfaces']:
+        if ':' in blk['peer_addr']:
+            continue
+        dut_port=blk['attachto']
+        port_id = mg_facts['minigraph_port_indices'][dut_port] 
+        neigh=mg_facts['minigraph_neighbors'][dut_port]       
+        if 'T0' in neigh['name']:
+             tor_addr[port_id]=blk['addr']
+             tor_peer_addr[port_id]=blk['peer_addr']
+        elif 'T2' in neigh['name']:
+             spine_addr[port_id]=blk['addr']
+             spine_peer_addr[port_id]=blk['peer_addr']
+            
+    for pid, dut_port in tor_ports.items():
+        port_id = mg_facts['minigraph_port_indices'][dut_port]
+        ansible_port='ansible_'+dut_port
+        tor_mac[port_id]=host_facts[ansible_port]['macaddress']
+    
+    for pid, dut_port in spine_ports.items():
+        port_id = mg_facts['minigraph_port_indices'][dut_port]
+        ansible_port='ansible_'+dut_port
+        spine_mac[port_id]=host_facts[ansible_port]['macaddress']
+
+    setup_information = {
+        'duthost': duthost,
+        'dut_tmp_dir': DUT_TMP_DIR,
+        'dst_ip_spine_blocked': '192.168.144.1',
+        'tor_ports': tor_ports,
+        'spine_ports': spine_ports,
+        'tor_addr': tor_addr,
+        'tor_peer_addr': tor_peer_addr,
+        'spine_addr': spine_addr,
+        'spine_peer_addr': spine_peer_addr,
+        'tor_ports_ids': tor_ports_ids,
+        'spine_ports_ids': spine_ports_ids,
+        'tor_mac': tor_mac,
+        'spine_mac': spine_mac,
+    }
+
+    logger.info('setup variables {}'.format(pprint.pformat(setup)))
+
+    # FIXME: There seems to be some issue with the initial setup of the ptfadapter, causing some of the
+    # TestBasicMPLS tests to fail because the forwarded packets are not being collected. This is an
+    # attempt to mitigate that issue while we continue to investigate the root cause.
+    #
+    # Ref: GitHub Issue #2032
+    logger.info("setting up the ptfadapter")
+    ptfadapter.reinit()
+
+    yield setup_information
+
+    duthost.command('rm -rf {}'.format(os.path.join(DUT_TMP_DIR, LABEL_POP_ROUTES)))
+    duthost.command('rm -rf {}'.format(os.path.join(DUT_TMP_DIR, LABEL_SWAP_ROUTES)))
+    duthost.command('rm -rf {}'.format(os.path.join(DUT_TMP_DIR, LABEL_PUSH_ROUTES)))
+    duthost.command('rm -rf {}'.format(os.path.join(DUT_TMP_DIR, LABEL_DEL_ROUTES)))
```

Comment:
since you are also cleaning up here how about rename fixture to 'setup_and_teardown'

Reply:
This is ony removing file from device. teardown as in removing mpls configuration is separate.